numba/tests/test_extending.py

```diff
@@ -1153,6 +1219,33 @@ def no_classmethod_in_base(nitems):
             str(raises.exception),
         )
 
+    def test_overload_callable_typeref(self):
+
+        @overload(CallableTypeRef)
+        def callable_type_call_ovld1(x):
+            if isinstance(x, types.Integer):
+                def impl(x):
+                    return 42.5 + x
+                return impl
+
+        @overload(CallableTypeRef)
+        def callable_type_call_ovld2(x):
+            if isinstance(x, (types.UnicodeType, types.StringLiteral)):
+                def impl(x):
+                    return '42.5' + x
+
+                return impl
+
+        typeref_type = get_callable_typeref(MyClass)
+
+        @njit
+        def foo(a, b):
+            return typeref_type(a), typeref_type(b)
```

Comment:
FYI, if you add 

```python
infer_global(MyClass, CallableTypeRef(MyClass))
```

the following can be supported:

```python
 @njit
def foo(a, b):
    return MyClass(a), MyClass(b)
```

Reply:
Thanks for pointing this! I didn't know `infer_global` can be used that way, so I used `typeof_impl.register(type)` same as it's done for Numba typed containers.